### DISCUSSION TITLE
Upgrading to Play 3.0.0 with Scala 3

### DISCUSSION THREAD

**Discussion by michbarsinai** [playframework/playframework#12226] [Q&A] [answered]
Hi Everyone, 
I have a Play/Scala application using play 2.8. The upgrade to play 2.9 and JDK21 went incredibly smooth - very impressive work by the Play core team. 

When I'm trying to upgrade to Scala 3, I get an error about SBT not finding the following dependencies:

```sbt
"org.playframework" % "play-slick" % "6.0.0",
"org.playframework" % "play-slick-evolutions" % "6.0.0",
"be.objectify" % "deadbolt-scala" % "2.8.2",
```

Any idea why these are missing for Scala3? I thought Scala 3 could use Scala 2.13 binaries, but I may be wrong. 

Thanks!

**Comment by Squeng** [comment]
Hi @michbarsinai 

I had run into the same problem with the MongoDB Scala driver (and ended up switching to the Java driver: https://www.mongodb.com/community/forums/t/scala-3-support/115681/7?u=paul_sevinc), so it seems like in general, libraries need to be explicitly built for Scala 3.

What's curious in your case is that Slick's pre-releases seem to have supported Scala 3:
![grafik](https://github.com/playframework/playframework/assets/7399173/aa8664e6-433a-443e-960b-25badb76a6e7)
I guess there's hope for Slick, but Deadbolt might be a dead end (no pun intended!).


Best of luck!
Paul

**Reply by mkurz** [reply, marked as answer]
play-slick is not published for Scala 3 yet. The pre-releases did use a slick milestone version, but the 6.0 final only published with Scala 2.13 support. So we have to wait for slick for an official Scala 3 supporting release.

Deadbolt will have a new version with Scala 3 support soon.

**Reply by michbarsinai** [reply]
Thanks! Good news about Deadbolt, too.

**Reply by mkurz** [reply]
@michbarsinai deadbolt-scala 2.9.0 and 3.0.0 released: 
* https://github.com/mkurz/deadbolt-2-scala/releases/tag/2.9.0
* https://github.com/mkurz/deadbolt-2-scala/releases/tag/3.0.0
